src/transformers/models/blip_2/modeling_blip_2.py

```diff
@@ -1149,7 +1326,7 @@ def forward(
             if type(encoder_attention_mask) == list:
                 encoder_extended_attention_mask = [self.invert_attention_mask(mask) for mask in encoder_attention_mask]
             elif encoder_attention_mask is None:
-                encoder_attention_mask = torch.ones(encoder_hidden_shape, device=device)
+                encoder_attention_mask = torch.ones(encoder_hidden_shape, device=device, dtype=torch.long)
```

Comment:
can you elaborate on why specifying torch.long is now needed?

Reply:
Thanks, I reverted the usage of `dtype=torch.long`.

I had some issues with fp16 when I started to work on this PR, and I believed that fixed the issue, now I am not able to replicate the it, even without `dtype=torch.long`, so it is better to remove it.